researchsuite-sdk/build.gradle

```diff
@@ -0,0 +1,46 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 26
+    buildToolsVersion "26.0.2"
+
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion 26
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+
+        javaCompileOptions {
+            annotationProcessorOptions {
+                arguments = ["room.schemaLocation": "$projectDir/schemas".toString()]
+            }
+        }
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile 'com.android.support:appcompat-v7:26.1.0'
+
+    compile 'org.sagebionetworks.researchsuite:api:0.0.3'
+
+    compile "android.arch.persistence.room:runtime:1.0.0-beta2"
```

Comment:
Room seems promising.  How well does it integrate/sync with the network when online/offline etc?  Will that be the responsibility of ResearchSuite, or will that somehow be tied to BridgeAndroidSDK?

Reply:
I'd like uploads and scheduled activities to use this model:
- ResearchSuite talks to to sqlite (putting this in BridgeSDK repo for now, but ideally it would be general enough to move to to ResearchSuite)
- [SyncAdapters](https://developer.android.com/training/sync-adapters/index.html) do the heavy lifting of syncing DB to REST API. This would live in BridgeSDK. The initial implementation of the SyncAdapters would call into the existing managers.